src/main/kotlin/no/nav/familie/ks/sak/kjerne/beregning/AndelGenerator.kt

```diff
@@ -1,22 +1,51 @@
 package no.nav.familie.ks.sak.kjerne.beregning
 
+import no.nav.familie.ks.sak.common.tidslinje.Tidslinje
+import no.nav.familie.ks.sak.common.tidslinje.utvidelser.kombinerMed
+import no.nav.familie.ks.sak.common.tidslinje.utvidelser.tilPerioderIkkeNull
+import no.nav.familie.ks.sak.kjerne.behandling.steg.vilkårsvurdering.domene.VilkårResultat
 import no.nav.familie.ks.sak.kjerne.behandling.steg.vilkårsvurdering.domene.Vilkårsvurdering
 import no.nav.familie.ks.sak.kjerne.beregning.domene.AndelTilkjentYtelse
 import no.nav.familie.ks.sak.kjerne.beregning.domene.TilkjentYtelse
+import no.nav.familie.ks.sak.kjerne.beregning.regelverkFørFebruar2025.tilAndelTilkjentYtelse
+import no.nav.familie.ks.sak.kjerne.personident.Aktør
 import no.nav.familie.ks.sak.kjerne.personopplysninggrunnlag.domene.Person
 import no.nav.familie.ks.sak.kjerne.regelverk.Regelverk
 import org.springframework.stereotype.Component
 
-interface AndelGenerator {
-    val regelverk: Regelverk
+abstract class AndelGenerator {
+    abstract val regelverk: Regelverk
 
-    fun beregnAndelerForBarn(
+    abstract fun beregnAndelerForBarn(
         søker: Person,
         barn: Person,
         vilkårsvurdering: Vilkårsvurdering,
         tilkjentYtelse: TilkjentYtelse,
     ): List<AndelTilkjentYtelse>
 
+    fun kombinerOgLagAndeler(
```

Comment:
Kunne man delt dette opp i to metoder `kombinerTidslinjer` og `lagAndeler`? Da får hver metode sitt eget ansvarsområde. Navnene er bare et forslag og kan sikkert forbedres 

Noe ala

```
interface AndelGenerator {
    val regelverk: Regelverk

    fun beregnAndelerForBarn(
        søker: Person,
        barn: Person,
        vilkårsvurdering: Vilkårsvurdering,
        tilkjentYtelse: TilkjentYtelse,
    ): List<AndelTilkjentYtelse>

    fun kombinerTidslinjer(
        søkersVilkårResultaterForskjøvetTidslinje: Tidslinje<List<VilkårResultat>>,
        barnetsVilkårResultaterForskjøvetTidslinje: Tidslinje<List<VilkårResultat>>,
    ) = barnetsVilkårResultaterForskjøvetTidslinje.kombinerMed(
        søkersVilkårResultaterForskjøvetTidslinje,
    ) { barnPeriode, søkerPeriode ->
        søkerPeriode?.let { barnPeriode }
    }

    fun lagAndeler(
        barnAktør: Aktør,
        tilkjentYtelse: TilkjentYtelse,
        tidslinje: Tidslinje<List<VilkårResultat>>,
    ): List<AndelTilkjentYtelse> =
        tidslinje
            .tilPerioderIkkeNull()
            .map { vilkårResultaterPeriode ->
                vilkårResultaterPeriode.tilAndelTilkjentYtelse(
                    tilkjentYtelse = tilkjentYtelse,
                    barnAktør = barnAktør,
                )
            }

    @Component
    class Lookup(
        private val andelGeneratorer: List<AndelGenerator>,
    ) {
        fun hentGeneratorForRegelverk(regelverk: Regelverk) = andelGeneratorer.single { it.regelverk == regelverk }
    }
}
```

Reply:
Hmm ulempen med det er at man eksponerer flere metoder fra interfacet som kanskje ikke egentlig burde være eksponert, mulig det er bedre å heller flytte de et annet sted

Reply:
Kanskje et `companion object` hadde vært bedre egnet her? 

```
interface AndelGenerator {

    // Code
    
    companion object {
        fun kombinerTidslinjer(
            søkersVilkårResultaterForskjøvetTidslinje: Tidslinje<List<VilkårResultat>>,
            barnetsVilkårResultaterForskjøvetTidslinje: Tidslinje<List<VilkårResultat>>,
        ) = barnetsVilkårResultaterForskjøvetTidslinje.kombinerMed(
            søkersVilkårResultaterForskjøvetTidslinje,
        ) { barnPeriode, søkerPeriode ->
            søkerPeriode?.let { barnPeriode }
        }

        fun lagAndeler(
            barnAktør: Aktør,
            tilkjentYtelse: TilkjentYtelse,
            tidslinje: Tidslinje<List<VilkårResultat>>,
        ): List<AndelTilkjentYtelse> =
            tidslinje
                .tilPerioderIkkeNull()
                .map { vilkårResultaterPeriode ->
                    vilkårResultaterPeriode.tilAndelTilkjentYtelse(
                        tilkjentYtelse = tilkjentYtelse,
                        barnAktør = barnAktør,
                    )
                }
    }
    
    // Code
    
}

```

Reply:
Syntes det hørtes lurt ut å flytte de til `companion object`. Da er "utility"-funksjonene fortsatt ganske nært knyttet til der de skal brukes. Alternativet er jo sikkert å flytte de helt ut i en egen fil eller noe, men synes det foreløpig fungerer godt å ha de i `companion object` også får vi se om vi ønsker å flytte de et annet sted dersom det gir mening senere 👍